### PR TITLE
20201216.1718 resolve feature 10783. tested and working.

### LINKED ISSUE #10783: [FEATURE] boxplot.py example: improve performance for larger datasets
### FOLLOW THESE INSTRUCTIONS CAREFULLY

*ISSUES THAT DO NOT CONTAIN NECESSARY INFORMATION MAY BE CLOSED, IMMEDIATELY*

The issue tracker is NOT the place for general support. For questions and technical assistance, come join the [Bokeh Project Discourse](https://discourse.bokeh.org).

**Is your feature request related to a problem? Please describe.**
I used the boxplot.py example to implement a very similar functionality in my project. The issue comes with the volume of the data: My dataset is several million rows.
Actually processing the data and getting a plot takes an extremely long time.

**Describe the solution you'd like**
I did some performance analysis and found that lines 30 to 35 are responsible for 95% of time spend:
```
if not out.empty:
    outx = []
    outy = []
    for keys in out.index:
        outx.append(keys[0])
        outy.append(out.loc[keys[0]].loc[keys[1]])
```

**Describe alternatives you've considered**
Suggesting to replace lines 30 to 35 with:
```
if not out.empty:
    outx = list(out.index.get_level_values(0)
    outy = list(out.values)
```

**Additional context**
analysis on my test dataset (15000 rows) did reduce this part alone from 6.66s to 4.14ms
running the code on my full dataset (35,278,061 rows) delivers a result in 3.46 seconds while I aborted the current code without result after 6 hours.

Happy to contribute.

### PR DESCRIPTION
**Pull request by Saravji**
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [X] issues: fixes #10783 
- [X] tests added / passed
- [N/A] release document entry (if new feature or API change)

### CODE REVIEW

File: examples/plotting/file/boxplot.py [hunk 1]
```diff
@@ -28,12 +28,9 @@ def outliers(group):
 
 # prepare outlier data for plotting, we need coordinates for every outlier.
 if not out.empty:
-    outx = []
-    outy = []
-    for keys in out.index:
-        outx.append(keys[0])
-        outy.append(out.loc[keys[0]].loc[keys[1]])
-
+    outx = list(out.index.get_level_values(0))
```
Comment (mattpap) [inline examples/plotting/file/boxplot.py:31]: Remove trailing whitespace to resolve code quality issues.

Comment (Saravji) [reply examples/plotting/file/boxplot.py:31]: You're fast! checks are still running....
White spaces already removed, waiting if more needs doing before pushing the changes.

Comment (Saravji) [reply examples/plotting/file/boxplot.py:31]: done

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks @Saravji

**Pull request merged**